src/backends/kafka/mod.rs

```diff
@@ -8,13 +8,42 @@ use rdkafka::config::{ClientConfig, RDKafkaLogLevel};
 use rdkafka::consumer::base_consumer::BaseConsumer;
 use rdkafka::consumer::{CommitMode, Consumer, ConsumerContext, Rebalance};
 use rdkafka::error::KafkaResult;
-use rdkafka::message::{Message, OwnedMessage};
+use rdkafka::message::{BorrowedHeaders, BorrowedMessage, Message, OwnedHeaders};
 use rdkafka::topic_partition_list::{Offset, TopicPartitionList};
 use std::collections::HashMap;
 use std::collections::HashSet;
 use std::sync::Mutex;
 use std::time::Duration;
 
+#[derive(Clone)]
+pub struct KafkaPayload {
+    pub key: Option<Vec<u8>>,
+    pub headers: Option<OwnedHeaders>,
+    pub payload: Option<Vec<u8>>,
+}
+
+fn create_kafka_message(msg: BorrowedMessage) -> ArroyoMessage<KafkaPayload> {
+    let topic = Topic {
+        name: msg.topic().to_string(),
+    };
+    let partition = Partition {
+        topic,
+        index: msg.partition() as u16,
+    };
+    let time_millis = msg.timestamp().to_millis().unwrap_or(0);
+
+    ArroyoMessage::new(
+        partition,
+        msg.offset() as u64,
+        KafkaPayload {
+            key: msg.key().map(|k| k.to_vec()),
+            headers: msg.headers().map(BorrowedHeaders::detach),
+            payload: msg.payload().map(|p| p.to_vec()),
```

Comment:
Isn't this making a copy of the payload ?
The previous version was not supposed to as `detach` is not supposed to copy anything, just changing the ownership.

Reply:
Yes it is, it is just making one less copy of topic, partition, offset and timestamp than previously.
If you look at the `detach` function it does actually make a copy of everything: https://github.com/fede1024/rust-rdkafka/blob/master/src/message.rs#L393-L408. This line is actually taken from the detach implementation.

Reply:
meh, I think we should find a different approach then (apart form this PR) to avoid the copy.
BorrowedMessage does not outlive the consumer. Most of the times the message does not outlive the consumer in our processing logic so the copy is useless.

Reply:
Mind creating an issue for that ?